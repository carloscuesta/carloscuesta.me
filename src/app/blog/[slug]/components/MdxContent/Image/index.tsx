import NextImage, { ImageProps } from 'next/image'

const Image = (props: unknown) => (
  <NextImage sizes="100vw" {...(props as ImageProps)} />
)

export default Image
