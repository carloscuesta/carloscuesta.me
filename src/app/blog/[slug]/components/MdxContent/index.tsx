import { MDXRemote } from 'next-mdx-remote/rsc'
import remarkGFM from 'remark-gfm'
import remarkToc from 'remark-toc'
import rehypeSlug from 'rehype-slug'
import rehypePrettyCode from 'rehype-pretty-code'
import rehypeExternalLinks from 'rehype-external-links'
import rehypeAutoLinkHeadings from 'rehype-autolink-headings'
import rehypeMinify from 'rehype-preset-minify'
import rehypeWrap from 'rehype-wrap-all'
import rehypeImageSize from 'rehype-probe-image-size'
import Image from './Image'

type Props = {
  source: string
}

const components = { img: Image } as const

const MdxContent = (props: Props) => (
  <div className="prose prose:pre prose-neutral relative max-w-full dark:prose-invert prose-headings:scroll-mt-16 prose-img:rounded-lg">
    <MDXRemote
      source={props.source}
      options={{
        mdxOptions: {
          remarkPlugins: [
            remarkGFM,
            [remarkToc, { tight: true, maxDepth: 4, ordered: true }],
          ],
          rehypePlugins: [
            rehypeSlug,
            // @ts-expect-error rehype-probe-image-size is not properly typed
            rehypeImageSize,
            [rehypeAutoLinkHeadings, { behavior: 'wrap' }],
            rehypeExternalLinks,
            [
              rehypePrettyCode,
              {
                theme: {
                  dark: require('sprinkles-vscode/themes/sprinkles-dark.json'),
                  light: require('sprinkles-vscode/themes/sprinkles-light.json'),
                },
                defaultLang: 'plaintext',
              },
            ],
            [rehypeWrap, { selector: 'table', wrapper: 'div.responsiveTable' }],
            rehypeMinify,
          ],
        },
      }}
      components={components}
    />
  </div>
)

export default MdxContent
