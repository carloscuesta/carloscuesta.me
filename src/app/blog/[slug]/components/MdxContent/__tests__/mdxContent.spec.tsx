import renderer from 'react-test-renderer'

import MdxContent from '../index'
import * as stubs from './stubs'

jest.mock('next-mdx-remote/rsc', () => ({
  MDXRemote: ({ source }: { source: string }) => <div>{source}</div>,
}))

describe('Header', () => {
  it('should match MdxContent component', () => {
    const wrapper = renderer.create(<MdxContent {...stubs.props} />)

    expect(wrapper).toMatchSnapshot()
  })
})
