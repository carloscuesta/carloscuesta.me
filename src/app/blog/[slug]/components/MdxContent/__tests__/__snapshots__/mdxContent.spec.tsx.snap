// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match MdxContent component 1`] = `
<div
  className="prose prose-neutral relative max-w-full dark:prose-invert prose-headings:scroll-mt-16 prose-img:rounded-lg"
>
  <div>
    
# Hello world
  
  </div>
</div>
`;
